packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -2926,6 +2927,9 @@ export class SnapController extends BaseController<
           this.#checkPhishingList.bind(this),
         );
         break;
+      case HandlerType.OnNameLookup:
+        assertStruct(result, OnNameLookupResponseStruct);
```

Comment:
This will disallow `null`, we still want to allow that.

Also it should be tested!